### DISCUSSION TITLE
D1 mini V4 with 2 sensors

### DISCUSSION THREAD

**Discussion by titou4307** [arendst/Tasmota#20493] [Q&A]
Hello
I have a D1 Mini V4 flashed with Tasmota Sensor firmware
I have connected on this HC-SR04P and there is no problem with this

I want to add a second sensor (DS18-B20) and a chose many GPIO but none appears in the screen web of the D1 mini
If I chose SI7021 so in the web menu it appears the 3 value possible

Any explication ?

**Comment by barbudor** [comment]
Crystal ball is in maintenance
Better to elaborate on what you have exactly done
A DS18x20 should work seamlessly with Tasmota sensors

**Reply by sfromis** [reply]
With DS18B20, you need to config the gpio specifically for `DS18x20`. Configuring as a completely different sensor type is not useful. Additionally, you should have a pull-up resistor connected, **or** set internal pullup with `SetOption74 1`

**Comment by titou4307** [comment]
Hello

I would like to have the level of my petrol tank into a local near my home and I want to have too the température

Here I have : 

![Tasmota 1](https://github.com/arendst/Tasmota/assets/56826797/47191935-62e0-4709-8039-283a7d638515)

![Tasmota 2](https://github.com/arendst/Tasmota/assets/56826797/fd250e36-f64a-4070-9ac7-c69c38d44958)

Sorry for my english porr (I am frenchee)

**Reply by barbudor** [reply]
There's no DS18X20 GPIO there
You need to configure one

**Reply by sfromis** [reply]
And remember pullup too, as mentioned.

**Comment by titou4307** [comment]
I have set like this with With DS18B20, and set "SetOption74 1"

![Tasmota 3](https://github.com/arendst/Tasmota/assets/56826797/0813457f-34e4-463f-8c1f-62461f22e53f)

Anything appears

![Tasmota 4](https://github.com/arendst/Tasmota/assets/56826797/636a4a3c-6130-4d0c-9317-c1f3cab5e355)

**Reply by sfromis** [reply]
Read the console right after boot, after first having set `Weblog 3` for details. With correct `DS18x20` config, you should see a message like:
`DSB: Sensors found 1`
Here, a value of 0 would indicate that the driver is active but did not detect a sensor. In that case, wiring may be an issue. And your use of gpio 14 should be fine.

**Comment by titou4307** [comment]
Hi
I will verify my connection, wiring and probe....
I will done this in the days and make return here

Thanks for the help
